src/Elastic.Transport/Diagnostics/TcpStats.cs

```diff
@@ -20,17 +20,30 @@ public static class TcpStats
 		/// Gets the active TCP connections
 		/// </summary>
 		/// <returns></returns>
-		public static TcpConnectionInformation[] GetActiveTcpConnections() =>
-			IPGlobalProperties.GetIPGlobalProperties().GetActiveTcpConnections();
+		public static TcpConnectionInformation[] GetActiveTcpConnections()
+		{
+			try
+			{
+				return IPGlobalProperties.GetIPGlobalProperties().GetActiveTcpConnections();
+			}
+			catch (NetworkInformationException) // host might not allow this information to be fetched.
+			{
+				// ignored
+			}
+
+			return null;			
+		}
 
 		/// <summary>
 		/// Gets the sum for each state of the active TCP connections
 		/// </summary>
 		public static ReadOnlyDictionary<TcpState, int> GetStates()
 		{
 			var states = new Dictionary<TcpState, int>(StateLength);
+
 			var connections = GetActiveTcpConnections();
```

Comment:
@russcam `EmptyReadOnly<TKey,TValue>.Dictionary` returns `IReadOnlyDictionary<TKey,TValue>`, but the signature on `GetStates` is `ReadOnlyDictionary`.

![image](https://user-images.githubusercontent.com/930343/102305896-2c4ff480-3fb5-11eb-83a8-2a3c0540bebd.png)

I can change to `IReadOnlyDictionary`, but.. that breaks a few things..

![image](https://user-images.githubusercontent.com/930343/102305970-50133a80-3fb5-11eb-99be-dc018083e5a6.png)

Happy to change, but wanted to get your OK first..

Reply:
can also do:

```
return new Dictionary<TcpState, int>(new Dictionary<TcpState, int>(0));
```

Which is a smaller change, but then there is some inconsistency, since we're not using that helper method..

Reply:
Ah, that's unfortunate. I think in next major, the method should be updated to return `IReadOnlyDictionary<TKey,TValue>`. Doing it in 7.x would be a breaking change so I think introducing a 

```
private static readonly ReadOnlyDictionary<TcpState, int> Empty = 
    new ReadOnlyDictionary<TcpState, int>(new Dictionary<TcpState, int>());
```

field and returning that would be OK.

Reply:
Agreed! Updating..

Reply:
Fixed.

Out of curiosity, how will you track updating the method signature in the next major? Should I create a _new_ issue to get that updated, tagged for the 8.x release, or is this something you guys will handle?

Reply:
I can open an issue to track that. We can take a PR for master without backporting to 7.x/7.10.

Reply:
@stevejgordon I'm happy to open a new PR for that if you like? Might need some guidance on the branching etc though, as not sure how a PR to master now will _not_ roll into the next 7.x release?